Tribler/Core/CacheDB/SqliteCacheDBHandler.py

```diff
@@ -862,6 +869,58 @@ def freeSpace(self, torrents2del):
         self._logger.info("Erased %d torrents", deleted)
         return deleted
 
+    def search_in_local_torrents_db(self, query, keys=None):
+        """
+        Search in the local database for torrents matching a specific query. This method also assigns a relevance
+        score to each torrent, based on the name, files and file extensions.
+        The algorithm is based on BM25. The document length factor is regarded since our "documents" are very small
+        (often a few keywords).
+        See https://en.wikipedia.org/wiki/Okapi_BM25 for more information about BM25.
+        """
+        search_results = []
+        keys_str = ", ".join(keys)
+        keywords = split_into_keywords(query, to_filter_stopwords=True)
+        infohash_index = keys.index('infohash')
+
+        results = self._db.fetchall("SELECT DISTINCT %s, Matchinfo(FullTextIndex, 'pcnalx') "
+                                    "FROM Torrent T, FullTextIndex "
+                                    "LEFT OUTER JOIN _ChannelTorrents C ON T.torrent_id = C.torrent_id "
+                                    "WHERE t.name IS NOT NULL AND t.torrent_id = FullTextIndex.rowid "
+                                    "AND C.deleted_at IS NULL AND FullTextIndex MATCH ?"
+                                    % keys_str, (" OR ".join(keywords),))
+
+        for result in results:
+            result = list(result)  # We convert the result to a mutable list since we have to decode the infohash
+            result[infohash_index] = str2bin(result[infohash_index])
+            matchinfo = result[len(keys)]  # The matchinfo is the last element in the results tuple
```

Comment:
No idea why I thought len wouldn't be a property of the datastructure. Probably a brain derp.